pkg/provider/gke/gke.go

```diff
@@ -426,6 +426,33 @@ func (c *GKE) nodePoolRunning(zone, projectID, clusterID, poolName string) (bool
 	return false, nil
 }
 
+// NodePoolCheck returns an error if any the nodepools do not exist.
+func (c *GKE) NodePoolCheck(*kingpin.ParseContext) error {
+	reqC := &containerpb.CreateClusterRequest{}
+
+	exists := true
+
+	for _, deployment := range c.gkeResources {
+		if err := yamlGo.UnmarshalStrict(deployment.Content, reqC); err != nil {
+			log.Fatalf("error parsing the cluster deployment file %s:%v", deployment.FileName, err)
+		}
+
+		for _, node := range reqC.Cluster.NodePools {
+			isRunning, err := c.nodePoolRunning(reqC.Zone, reqC.ProjectId, reqC.Cluster.Name, node.Name)
+			if err != nil {
+				log.Fatalln("error fetching nodePool info")
+			}
+			exists = exists && isRunning
```

Comment:
exists is always true, why is it needed?

Reply:
if any of the nodepool is running then `isRunning` will be `false` and then `exists` will be false and will continue to be false till the return statement below.

I updated the function description, now it's ` // NodePoolCheck returns an error if any the nodepools do not exist. `

Reply:
^ nodepool is not running